pkg/systemd/quadlet/quadlet.go

```diff
@@ -1248,6 +1285,149 @@ func ConvertImage(image *parser.UnitFile) (*parser.UnitFile, string, error) {
+		return nil, "", err
+	}
+
+	podman := createBasePodmanCommand(build, BuildGroup)
+	podman.add("build")
+
+	stringKeys := map[string]string{
+		KeyArch:     "--arch",
+		KeyAuthFile: "--authfile",
+		KeyPull:     "--pull",
+		KeyTarget:   "--target",
+		KeyVariant:  "--variant",
+	}
+
+	boolKeys := map[string]string{
+		KeyTLSVerify: "--tls-verify",
+		KeyForceRM:   "--force-rm",
+	}
+
+	for key, flag := range stringKeys {
+		lookupAndAddString(build, BuildGroup, key, flag, podman)
+	}
+
+	for key, flag := range boolKeys {
+		lookupAndAddBoolean(build, BuildGroup, key, flag, podman)
+	}
+
+	annotations := build.LookupAllKeyVal(BuildGroup, KeyAnnotation)
+	podman.addAnnotations(annotations)
+
+	dns := build.LookupAll(BuildGroup, KeyDNS)
+	for _, ipAddr := range dns {
+		podman.addf("--dns=%s", ipAddr)
+	}
+
+	dnsOptions := build.LookupAll(BuildGroup, KeyDNSOption)
+	for _, dnsOption := range dnsOptions {
+		podman.addf("--dns-option=%s", dnsOption)
+	}
+
+	dnsSearches := build.LookupAll(BuildGroup, KeyDNSSearch)
+	for _, dnsSearch := range dnsSearches {
+		podman.addf("--dns-search=%s", dnsSearch)
+	}
+
+	podmanEnv := build.LookupAllKeyVal(BuildGroup, KeyEnvironment)
+	podman.addEnv(podmanEnv)
+
+	groupsAdd := build.LookupAll(BuildGroup, KeyGroupAdd)
+	for _, groupAdd := range groupsAdd {
+		if len(groupAdd) > 0 {
+			podman.addf("--group-add=%s", groupAdd)
+		}
+	}
+
+	labels := build.LookupAllKeyVal(BuildGroup, KeyLabel)
+	podman.addLabels(labels)
+
+	builtImageName, ok := names[build.Filename]
+	if !ok {
+		return nil, "", fmt.Errorf("no ImageTag key specified")
+	}
+
+	podman.addf("--tag=%s", builtImageName)
+
+	addNetworks(build, BuildGroup, service, names, podman)
+
+	secrets := build.LookupAllArgs(BuildGroup, KeySecret)
+	for _, secret := range secrets {
+		podman.add("--secret", secret)
+	}
+
+	if err := addVolumes(build, service, BuildGroup, names, podman); err != nil {
+		return nil, "", err
+	}
+
+	// In order to build an image locally, we need either a File key pointing directly at a
+	// Containerfile, or we need a context or WorkingDirectory containing all required files.
+	// SetWorkingDirectory= can also be a path, a URL to either a Containerfile, a Git repo, or
+	// an archive.
+	context, err := handleSetWorkingDirectory(build, service, BuildGroup)
+	if err != nil {
+		return nil, "", err
+	}
+
+	workingDirectory, okWD := service.Lookup(ServiceGroup, ServiceKeyWorkingDirectory)
+	filePath, okFile := build.Lookup(BuildGroup, KeyFile)
+	if (!okWD || len(workingDirectory) == 0) && (!okFile || len(filePath) == 0) && len(context) == 0 {
+		return nil, "", fmt.Errorf("neither SetWorkingDirectory, nor File key specified")
+	}
+
+	if len(filePath) > 0 {
+		podman.addf("--file=%s", filePath)
+	}
+
+	handlePodmanArgs(build, BuildGroup, podman)
+
+	// Context or WorkingDirectory has to be last argument
+	if len(context) > 0 {
+		podman.add(context)
+	} else if !filepath.IsAbs(filePath) && !isHttpUrl(filePath) {
```

Comment:
I'd suggest we use the same logic in `podman build` [here][1], which handles more types of URL.

[1]: https://github.com/containers/podman/blob/34d2d20d46c1ad3cf13432fc481047f1b285ff20/cmd/podman/common/build.go#L206.

Reply:
I have to take a closer look at this if it's just adding a bunch more checks for different URL types (`git://` def. makes sense). I didn't do this yet, because the postman build docs explicitly only mention http(s) for `--file`. But I am curious why people can't just put a https:// in front of their GitHub URLs? Why the special treatment of GitHub which just doesn't look like a URL, and thus could be misdetected as a relative path?

Reply:
For now I have expanded the regex check I added to include github.com/ and git:// links which seems to work according to the e2e tests. Please let me know if you want me to follow the `podman build` code even more closely.

Reply:
Looks good to me!